_posts/2023-07-14-trino-fest-2023-dune.md

```diff
@@ -0,0 +1,59 @@
+---
+layout: post
+title: "DuneSQL - A query engine for blockchain data"
+author: "Miguel Filipe, Jonas Irgens Kylling, Ryan Duan"
+excerpt_separator: <!--more-->
+image: /assets/blog/trino-fest-2023/Dune.png
+show_pagenav: false
+---
+
+The need to make blockchain data easily accessible has risen over the recent
+years due to the rise of cryptocurrencies. Thus, Dune has made this their
+mission: to make crypto data accessible. Dune is a community data platform for
+querying public blockchain data and building beautiful dashboards. They use
+their own query engine called DuneSQL, a fork of TrinoSQL, to query blockchain
+data. In Miguel and Jonas' talk, they talk about the challenges in querying
+blockchain data, their transition to Trino, and how DuneSQL is operated. Give
+the talk a listen or keep reading for a recap.
+
+<!--more-->
+
+{% youtube sCJncarnGdU %}
+
+## Recap
+
+A community data platform is a serverless, open access, community wide
+collaboration portal or application that Dune uses for their query engine.
+However, they have some difficulties with blockchain data such as processing and
+ingesting raw data, deserializing and decoding function calls and arguments, and
+allowing the community to build abstractions on top. In addition, their engine
+has 10000s of queries that are executed and saved and re-used each day.
+
+At first, Dune used PostgreSQL where they sharded per blockchain and used a
+vertical scaling. However, they quickly ran into bottleneck issues on storage
+size and IOPS. Thus, they switched to Spark/Databricks for a horizontal scaling
```

Comment:
IOPS probably needs to be spelled out, maybe in parentheses. Not an acronym you encounter super frequently.